metals/src/main/scala/scala/meta/internal/metals/watcher/FileWatcher.scala

```diff
@@ -152,7 +152,8 @@ object FileWatcher {
         // However, the events are then filtered to receive only relevant events
 
         val trie = PathTrie(
-          pathsToWatch.files ++ pathsToWatch.directories
+          pathsToWatch.files ++ pathsToWatch.directories,
+          workspace,
```

Comment:
Could you explain why/for what cases we need this change?

Reply:
This is not related only to bazel-bsp, but I noticed this bug while working on it. I don't remember the exact stack trace, but when bazel-bsp fails because of invalid config, the only build target uri we get is `bsp-workspace-root` (its not being a valid uri might cause some issues, but I haven't noticed any yet), and request for sources returns
```
"items": [
    {
      "target": {
        "uri": "bsp-workspace-root"
      },
      "sources": [],
      "roots": []
    }
  ]
  ```
  We then tried to create `PathTrie` with empty `paths`, which resulted in watching the whole file system.
  Basically `PathTrie` should be used as optimisation to not watch the whole workspace directory, so if the paths are empty, we fallback to watching workspace directory.

Reply:
Thanks for the explanation!
What do you think about adding these explanation to code?